server/entity/projectile.go

```diff
@@ -221,6 +221,11 @@ func (lt *ProjectileBehaviour) tryPickup(e *Ent) {
 		if !ok {
 			continue
 		}
+
+		if collector.Collect(lt.conf.PickupItem) == 0 {
```

Comment:
This breaks collecting the projectile when the item stack is empty, which will be intentional behavior for projectiles shot by a skeleton, drowned, etc.